packages/typespec-azure-resource-manager/lib/models.tsp

```diff
@@ -293,7 +293,7 @@ model ResourceKindProperty {
  */
 @doc("The response of a {name} list operation.", Resource)
 @friendlyName("{name}ListResult", Resource)
-model ResourceListResult<Resource extends Foundations.Resource> is Azure.Core.Page<Resource>;
+model ResourceListResult<Resource extends {}> is Azure.Core.Page<Resource>;
```

Comment:
There is some issue with this. If the intension has been to support scalar or array types, it would not work. [Playground](https://azure.github.io/typespec-azure/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3Jl3yvIK3Jlc291cmNlLW1hbmFnZXIiOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TRFVbpAKnIG0HEZS5Db3JlzhJSx3dNxnY7Cgptb2RlbCBDdXN0b23IHkxpc3RSZXN1bHQ8yBMgZXh0ZW5kcyB7fT4KICBpc8tjLlBhZ2XJKj47CgpAcm91dGUoIi8iKQpAZ2V0Cm9wIGdldCgpOtpqc3RyaW5nzT4xzj8x30DEQFtdPjsK&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-rulesets%2Fresource-manager%22%5D%7D%7D). In addition, `Azure.Core.Page<T extends Model>`, making this would fail in Azure.Core as well.

Reply:
`{}` and `Model` are equivalent constraints.  Also, this handles *most* list cases.  If we want to open up the paged data to be any type (or scalar and model types), we can use our own custom page.  [Here is a playground](https://azure.github.io/typespec-azure/playground?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7CtIZcmVzdNUZdmVyc2lvbmluZ8wfYXp1cmUtdG9vbHMvyCstxhVjb3Jl3yvIK3Jlc291cmNlLW1hbmFnZXIiOwoKdXNpbmcgVHlwZVNwZWMuSHR0cDvQFVJlc3TRFVbpAKnIG0HEZS5Db3JlzhJSx3dNxnY7CgpAdXNlRGVwZW5kZW5jeSjKOchWcy52MV8wX1ByZXZpZXdfMikKQHNlcnZpY2UoewogIHRpdGxlOiAiU2FtcGxlIiwKfSkKbmFtZXNwYWNlIE1pY3Jvc29mdC5Gb287Cm1vZGVsIEN1c3RvbegAi0xpc3RSZXN1bHQ8yBMgZXh0ZW5kcyB7fT4KICBpc%2BsA0C5QYWdlySo%2B5ADCcGFnZWTGQ8diQW551l91bHQ%2BIOQAt0BpdGVtcwogIHZhbHVlOiBBcnJhecgiOwrEIm5leHRMaW5rCiAgyAs%2FOiB1cmw7Cn0KCkByb3V0ZSgiLyIpCkBnZXQKb3AgZ2V0KCk6%2BgDgxHnkARw6IHN0cuUBrn3EbMlLMc5MMd9NICBpZNhLMs5LMsRL9gEWxkA%2BOwo%3D&e=%40azure-tools%2Ftypespec-autorest&options=%7B%22linterRuleSet%22%3A%7B%22extends%22%3A%5B%22%40azure-tools%2Ftypespec-azure-rulesets%2Fresource-manager%22%5D%7D%7D) showing the definition for ResourceListResult working with `Page<T>` and a completely custom paged result.

Reply:
Mark for this approach, are you thinking on creating a new operation or do we still see this as an update on ArmResourceListByParent operation? 
if we update on ArmResourceListByParent, is there a way to do it without being a breaking change?